src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,54 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+
 public class SalaryInfo {
+    private static final String FIRST_LINE_FORMAT = "Report for period %s - %s";
+    private static final String SALARY_LINE_FORMAT = "\n%s - %d";
+    private static final int DEFAULT_SALARY_VALUE = 0;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate dateFromLocal = LocalDate.parse(dateFrom, SalaryData.DATE_TIME_FORMATTER);
+        LocalDate dateToLocal = LocalDate.parse(dateTo, SalaryData.DATE_TIME_FORMATTER);
+        DateRange dateRange = new DateRange(dateFromLocal, dateToLocal);
+        Map<String, Integer> nameSalaryMap = getNameSalaryMap(names, data, dateRange);
+        StringBuilder sb = new StringBuilder(String.format(FIRST_LINE_FORMAT, dateFrom, dateTo));
+        for (String name : names) {
+            int salary = nameSalaryMap.getOrDefault(name, DEFAULT_SALARY_VALUE);
+            sb.append(String.format(SALARY_LINE_FORMAT, name, salary));
+        }
+        return sb.toString();
+    }
+
+    private static Map<String, Integer> getNameSalaryMap(String[] names,
+                                                         String[] data,
+                                                         DateRange dateRange) {
+        return Arrays.stream(data)
```

Comment:
Don’t use two-dimension arrays and HashMap(or any other Map). As per the checklist, avoid using Maps. You can achieve the same result by iterating through the data and calculating the salaries manually.